src/account/account-data.ts

```diff
@@ -1,32 +1,64 @@
-import { Wallet, utils } from 'ethers'
-import { assertMnemonic, assertPassword, assertUsername, removeZeroFromHex } from './utils'
+import { utils, Wallet } from 'ethers'
+import { assertMnemonic, assertPassword, assertRegistrationAccount, assertUsername, removeZeroFromHex } from './utils'
 import { prepareEthAddress } from '../utils/address'
 import { getEncryptedMnemonic } from './mnemonic'
 import { decryptText } from './encryption'
-import { uploadPortableAccount, downloadPortableAccount } from './account'
+import { downloadPortableAccount, uploadPortableAccount, UserAccountWithMnemonic } from './account'
 import { Connection } from '../connection/connection'
 import { AddressOptions, isAddressOptions, isMnemonicOptions, MnemonicOptions } from './types'
-import { ENS } from '@fairdatasociety/fdp-contracts'
-import { Utils } from '@ethersphere/bee-js'
+import { ENS, PublicKey } from '@fairdatasociety/fdp-contracts'
+import { Reference, Utils } from '@ethersphere/bee-js'
+import CryptoJS from 'crypto-js'
+import { bytesToHex } from '../utils/hex'
 
 export class AccountData {
   /**
    * Active FDP account wallet
    */
-  public wallet?: Wallet
+  public wallet?: utils.HDNode
+
+  /**
+   * Active FDP account's seed for entity creation
+   */
   public seed?: Uint8Array
 
+  /**
+   * Public key for FDP account creation
+   */
+  public publicKey?: PublicKey
+
   constructor(public readonly connection: Connection, public readonly ens: ENS) {}
 
   /**
-   * Sets the current account's wallet to interact with the data
+   * Connects wallet with ENS
+   */
+  private connectWalletWithENS(seed: Uint8Array) {
+    this.seed = seed
+    this.wallet = utils.HDNode.fromSeed(seed).derivePath(`m/44'/60'/0'/0/0`)
+    this.ens.connect(new Wallet(this.wallet!.privateKey).connect(this.ens.provider))
+  }
+
+  /**
+   * Sets FDP account from seed
+   *
+   * With the help of the seed, account data can be managed, but cannot register a new account
    *
-   * @param wallet BIP-039 + BIP-044 Wallet
+   * @param seed data extracted from mnemonic phrase or from uploaded account
    */
-  setActiveAccount(wallet: Wallet, seed?: Uint8Array): void {
-    this.seed = seed || new Uint8Array()
-    this.wallet = wallet.connect(this.ens.provider)
-    this.ens.connect(this.wallet)
+  setAccountFromSeed(seed: Uint8Array): void {
+    // correct public key can't be extracted from seed
```

Comment:
Can public key be extracted after a wallet is created in the `connectWalletWithENS` function?

Reply:
It can be received. But the public key obtained from hdwallet (which is generated from seed) and from mnemonic have a different length. The public key from hdwallet cannot be set when registering ens, I get an error.